pkg/systemd/overview-cards/lastLogin.jsx

```diff
@@ -0,0 +1,105 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React, { useState, useEffect } from 'react';
+import { Flex, FlexItem } from '@patternfly/react-core';
+import { ExclamationTriangleIcon, UserIcon } from "@patternfly/react-icons";
+
+import cockpit from "cockpit";
+import { LoginMessage } from "./loginMessage.jsx";
+
+import './lastLogin.scss';
+
+const _ = cockpit.gettext;
+
+const LastLogin = () => {
+    const [messages, setLoginMessages] = useState(null);
+    const [username, setUsername] = useState(null);
+
+    useEffect(() => {
+        if (messages === null) {
+            const bridge = cockpit.dbus(null, { bus: "internal" });
+            bridge.call("/LoginMessages", "cockpit.LoginMessages", "Get", [])
+                    .then(reply => {
+                        const obj = JSON.parse(reply[0]);
+                        if (obj.version == 1) {
+                            if (obj['fail-count'] > 5) {
+                                obj.type = "danger";
+                            } else if (obj['fail-count'] > 0) {
+                                obj.type = "warning";
+                            } else {
+                                obj.type = "info";
+                            }
+
+                            setLoginMessages(obj);
+                        } else {
+                        // empty reply is okay -- older bridges just don't send that information
+                            if (obj.version !== undefined)
+                                console.error("unknown login-messages:", reply[0]);
+                        }
+                    })
+                    .catch(error => {
+                        console.error("failed to fetch login messages:", error);
+                    });
+        }
+        if (username === null) {
+            cockpit.user().then(user => setUsername(user.name));
+        }
+    }, [messages, username]);
+
+    if (messages === null || !messages['last-login-time'] || username === null) {
+        return null;
+    }
+
+    let icon = null;
+    let text = null;
+    const failedLogins = messages['fail-count'];
+
+    if (failedLogins) {
+        icon = <ExclamationTriangleIcon className="system-information-last-login-warning-icon" size="sm" />;
+        text = (
+            <FlexItem id="system_last_login"
+                    className="pf-u-text-break-word system-information-failed-login"
+                    spacer={{ default: 'spacerSm' }}>
+                {cockpit.format(cockpit.ngettext("$0 failed login attempt", "$0 failed login attempts", failedLogins), failedLogins)}
+            </FlexItem>
+        );
+    } else {
+        icon = <UserIcon className="system-information-last-login-icon" size="sm" />;
```

Comment:
Last login is shown only when there is no failed one? I thought that we would always show last login and if tehre is failed one, then also show that one

Reply:
I also asked about this - seems that the design needed the oppositite. @garrett can you clarify this?

Reply:
The design assumes that there would be a list of logins on the user page, but we can add both (at least for now, possibly even after) if that would make everyone happier.